vignettes/adsl.Rmd

```diff
@@ -293,68 +293,65 @@ to a suitable `EOSSTT` value.
 The main reason for discontinuation is usually stored in `DSDECOD` while `DSTERM` provides 
 additional details regarding subject’s discontinuation (e.g., description of `"OTHER"`).
 
-The function `derive_vars_disposition_reason()` can be used to derive a disposition reason (along with 
-the details, if  required) at a specific timepoint. 
-The relevant disposition variable(s) (`DS.DSDECOD`, `DS.DSTERM`) are selected by adjusting the 
-filter parameter and used to derive the main reason (and details).
+The functions `restrict_derivation` and `derive_vars_merged()` can be used to derive a disposition reason (along with the details, if required) at a specific timepoint. 
+The relevant disposition variable(s) (`DS.DSDECOD`, `DS.DSTERM`) are selected by adjusting the filter parameter and used to derive the main reason (and details).
 
-To derive the End of Study reason(s) (`DCSREAS` and `DCSREASP`), the call would be:
+To derive the End of Study reason(s) (`DCSREAS` and `DCSREASP`), the function will map 
 
-```{r eval=TRUE}
-adsl <- adsl %>%
-  derive_vars_disposition_reason(
-    dataset_ds = ds,
-    new_var = DCSREAS,
-    reason_var = DSDECOD,
-    new_var_spe = DCSREASP,
-    reason_var_spe = DSTERM,
-    filter_ds = DSCAT == "DISPOSITION EVENT" & DSDECOD != "SCREEN FAILURE"
-  )
+- `DCSREAS` as `DSDECOD` if `DSDECOD` is not `"COMPLETED"`, `SCREEN FAILURE`, or `NA`, `NA` otherwise
+- `DCSREASP` as `DSTERM` if `DSDECOD` is not `"COMPLETED"`, `SCREEN FAILURE`, or `NA`, `NA` otherwise
+
+```{r}
+adsl <- restrict_derivation(
```

Comment:
I think we do not need `restrict_derivation()` as `USUBJID` should not be `NA`.

Maybe we should also remove `restrict_derivation()` from the deprecation message. I have mentioned it in the issue description but I do not remember why. It seems that it is not needed.